### DISCUSSION TITLE
Problems with Gateway VM communication to Cloud

### DISCUSSION THREAD

**Discussion by tayloreisman16** [magma/magma#7736] [Q&A] [answered]
Hey guys, I'm following the magma instructions here: 
[Magma Install](https://magma.github.io/magma/docs/basics/quick_start_guide)
And it is failing on this step:
![image](https://user-images.githubusercontent.com/18565850/123166300-dfa11680-d43a-11eb-8342-ff3a075c40e4.png)
It's failing, however, with making a connection:
`  File "/usr/lib/python3/dist-packages/requests/sessions.py", line 646, in send
    r = adapter.send(request, **kwargs)
  File "/usr/lib/python3/dist-packages/requests/adapters.py", line 498, in send
    raise ConnectionError(err, request=request)
requests.exceptions.ConnectionError: ('Connection aborted.', PermissionError(13, 'Permission denied'))
`
As you can see, a ConnectionError.

Solutions I've tried:
Adding $USER to the docker group.
Searched for answers online.

I'm pretty sure one of my connections in 'ifconfig' is incorrectly set up. However, when I look at ifconfig, it is a mess. I'll paste it below.

`br-7584a08df37d: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 1500
        inet 172.18.0.1  netmask 255.255.0.0  broadcast 172.18.255.255
        inet6 fe80::42:1eff:fe24:9c38  prefixlen 64  scopeid 0x20<link>
        ether 02:42:1e:24:9c:38  txqueuelen 0  (Ethernet)
        RX packets 5369  bytes 1176863 (1.1 MB)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 6531  bytes 1972950 (1.9 MB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

docker0: flags=4099<UP,BROADCAST,MULTICAST>  mtu 1500
        inet 172.17.0.1  netmask 255.255.0.0  broadcast 172.17.255.255
        ether 02:42:4d:b8:e8:04  txqueuelen 0  (Ethernet)
        RX packets 0  bytes 0 (0.0 B)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 0  bytes 0 (0.0 B)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

enp0s31f6: flags=4099<UP,BROADCAST,MULTICAST>  mtu 1500
        ether b0:6e:bf:cd:b8:db  txqueuelen 1000  (Ethernet)
        RX packets 0  bytes 0 (0.0 B)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 0  bytes 0 (0.0 B)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0
        device interrupt 16  memory 0xdf600000-df620000  

lo: flags=73<UP,LOOPBACK,RUNNING>  mtu 65536
        inet 127.0.0.1  netmask 255.0.0.0
        inet6 ::1  prefixlen 128  scopeid 0x10<host>
        loop  txqueuelen 1000  (Local Loopback)
        RX packets 33430  bytes 4946770 (4.9 MB)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 33430  bytes 4946770 (4.9 MB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

vboxnet2: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 1500
        inet 192.168.60.1  netmask 255.255.255.0  broadcast 192.168.60.255
        inet6 fe80::800:27ff:fe00:2  prefixlen 64  scopeid 0x20<link>
        ether 0a:00:27:00:00:02  txqueuelen 1000  (Ethernet)
        RX packets 0  bytes 0 (0.0 B)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 123  bytes 14502 (14.5 KB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

vboxnet3: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 1500
        inet 192.168.129.1  netmask 255.255.255.0  broadcast 192.168.129.255
        inet6 fe80::800:27ff:fe00:3  prefixlen 64  scopeid 0x20<link>
        ether 0a:00:27:00:00:03  txqueuelen 1000  (Ethernet)
        RX packets 0  bytes 0 (0.0 B)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 120  bytes 14334 (14.3 KB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

veth26d0ee0: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 1500
        inet6 fe80::54c5:2fff:fe70:42b1  prefixlen 64  scopeid 0x20<link>
        ether 56:c5:2f:70:42:b1  txqueuelen 0  (Ethernet)
        RX packets 25087  bytes 12696235 (12.6 MB)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 39461  bytes 9941928 (9.9 MB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

veth3642e10: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 1500
        inet6 fe80::1c14:fcff:fe8e:837  prefixlen 64  scopeid 0x20<link>
        ether 1e:14:fc:8e:08:37  txqueuelen 0  (Ethernet)
        RX packets 6946  bytes 1220131 (1.2 MB)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 7403  bytes 1971180 (1.9 MB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

veth8d0e0cc: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 1500
        inet6 fe80::e8ac:80ff:fe9a:858d  prefixlen 64  scopeid 0x20<link>
        ether ea:ac:80:9a:85:8d  txqueuelen 0  (Ethernet)
        RX packets 12021  bytes 1258550 (1.2 MB)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 17279  bytes 1682149 (1.6 MB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

vethb0ad45f: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 1500
        inet6 fe80::9437:e9ff:fea9:ff64  prefixlen 64  scopeid 0x20<link>
        ether 96:37:e9:a9:ff:64  txqueuelen 0  (Ethernet)
        RX packets 0  bytes 0 (0.0 B)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 132  bytes 17562 (17.5 KB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

vethc16997f: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 1500
        inet6 fe80::e8da:82ff:fe2e:d5a1  prefixlen 64  scopeid 0x20<link>
        ether ea:da:82:2e:d5:a1  txqueuelen 0  (Ethernet)
        RX packets 13814  bytes 1310949 (1.3 MB)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 10492  bytes 1077965 (1.0 MB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

vethfb5dfdd: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 1500
        inet6 fe80::e4b1:8cff:fe0e:b672  prefixlen 64  scopeid 0x20<link>
        ether e6:b1:8c:0e:b6:72  txqueuelen 0  (Ethernet)
        RX packets 0  bytes 0 (0.0 B)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 134  bytes 17773 (17.7 KB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

vethfe1ca57: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 1500
        inet6 fe80::a0de:f0ff:fe7d:4ca5  prefixlen 64  scopeid 0x20<link>
        ether a2:de:f0:7d:4c:a5  txqueuelen 0  (Ethernet)
        RX packets 39343  bytes 9925144 (9.9 MB)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 25190  bytes 12709437 (12.7 MB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

virbr0: flags=4099<UP,BROADCAST,MULTICAST>  mtu 1500
        inet 192.168.122.1  netmask 255.255.255.0  broadcast 192.168.122.255
        ether 52:54:00:d5:9c:c5  txqueuelen 1000  (Ethernet)
        RX packets 0  bytes 0 (0.0 B)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 0  bytes 0 (0.0 B)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

wlp5s0: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 1500
        inet 10.3.0.198  netmask 255.255.248.0  broadcast 10.3.7.255
        inet6 fe80::a745:b5b7:272d:bf26  prefixlen 64  scopeid 0x20<link>
        ether 64:6e:69:6a:e9:c9  txqueuelen 1000  (Ethernet)
        RX packets 1658208  bytes 1548166026 (1.5 GB)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 368309  bytes 70368658 (70.3 MB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0`

Does anybody have any ideas? I appreciate the help! I'm sure this is something simple, but I have followed the installation instructions to the T, not sure why it isn't working.

**Comment by tayloreisman16** [comment, marked as answer]
Turns out it was a permissions issue on the magma directory. I needed to chown -R [user] ~/magma.

**Reply by dzakwanf** [reply]
step you do not need to do, go to console register your vm to nms using hardware id on vm agw
